lib/nuget-parser/nuspec-parser.ts

```diff
@@ -1,43 +1,42 @@
-'use strict';
-const JSZip = require('jszip');
-const fs = require('fs');
-const path = require('path');
-const parseXML = require('xml2js').parseString;
-const Dependency = require('./dependency');
-const _ = require('lodash');
-const debug = require('debug')('snyk');
+import * as JSZip from 'jszip';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as parseXML from 'xml2js';
+import * as dependency from './dependency';
+import * as _ from 'lodash';
+import * as debugModule from 'debug';
+const debug = debugModule('snyk');
 
 const targetFrameworkRegex = /([.a-zA-Z]+)([.0-9]+)/;
 
-function parseNuspec(dep, targetFramework, sep) {
+export async function parseNuspec(dep, targetFramework) {
   return Promise.resolve()
-    .then(function () {
-      const pathSep = sep || '.';
+    .then(() => {
       const nupkgPath =
-        path.resolve(dep.path, dep.name + pathSep + dep.version + '.nupkg');
+        path.resolve(dep.path, dep.name + '.' + dep.version + '.nupkg');
       const nupkgData = fs.readFileSync(nupkgPath);
       return JSZip.loadAsync(nupkgData);
     })
-    .then(function (nuspecZipData) {
-      const nuspecFile = Object.keys(nuspecZipData.files).find(function (file) {
+    .then((nuspecZipData) => {
+      const nuspecFiles = Object.keys(nuspecZipData.files).filter((file) => {
         return (path.extname(file) === '.nuspec');
       });
-      return nuspecZipData.files[nuspecFile].async('string');
+      return nuspecZipData.files[nuspecFiles[0]].async('string');
     })
-    .then(function (nuspecContent) {
-      return new Promise(function (resolve, reject) {
-        parseXML(nuspecContent, function (err, result) {
+    .then((nuspecContent) => {
+      return new Promise((resolve, reject) => {
+        parseXML.parseString(nuspecContent, (err, result) => {
           if (err) {
             return reject(err);
           }
 
-          let ownDeps = [];
+          let ownDeps: any = [];
           // We are only going to check the first targetFramework we encounter
           // in the future we may want to support multiple, but only once
           // we have dependency version conflict resolution implemented
-          // _(targetFrameworks).forEach(function (targetFramework) {
-          _(result.package.metadata).forEach(function (metadata) {
-            _(metadata.dependencies).forEach(function (rawDependency) {
+          // _(targetFrameworks).forEach((targetFramework) => {
```

Comment:
Please delete this code if it won't be used

Reply:
removed